File: days/day_17/task_2.py
from helpers import read_input
from days.day_17.task_1 import tick


def run(data):
    cubes = {}

    for y, row in enumerate(data):
        for x, column in enumerate(row):
            cubes[(x, y, 0, 0)] = column

    for _ in range(6):
        cubes = tick(cubes)

    return sum([1 if state == '#' else 0 for state in cubes.values()])


if __name__ == "__main__":
    print(run(read_input('input.txt')))


def test_input():
    assert run(read_input('input.txt')) == 2064


def test_test():
    assert run(read_input('test.txt')) == 848

File: src/days/day6/task2.ts
import esMain from "es-main";
import { findMarkerPosition, parseInput } from "./task1";

if (esMain(import.meta)) {
  console.log("TEST:", findMarkerPosition(await parseInput("./test.txt"), 14));
  console.log(
    "INPUT:",
    findMarkerPosition(await parseInput("./input.txt"), 14)
  );
}
